lib/flux/settings-reducer.js

```diff
@@ -49,11 +49,25 @@ const markdownEnabled = ( state = false, action ) => {
 	return action.markdownEnabled;
 };
 
+const sourceList = ( state = { width: 330, isResizing: false, canResize: false }, { type, width, canResize, isResizing } ) => {
+	switch ( type ) {
+		case 'setSourceListCanResize':
+			return Object.assign( {}, state, { canResize } )
+		case 'setSourceListResizing':
+			return Object.assign( {}, state, { isResizing } )
+		case 'setSourceListWidth':
+			return Object.assign( {}, state, { width } );
+		default:
+			return state;
+	}
+};
```

Comment:
This could probably be its own composed reducer, since we're starting to bundle together different logic. I realize that duplicate code is becoming a problem here, so maybe we need a helper function.

```js
import {
	equals,
	ifElse,
	prop
} from 'ramda';

const simpleState = ( type, propName, default ) => 
	( state = default, action ) =>
		ifElse( equals( type, prop( 'type', action ) ), prop( propName, action ), state );

const sourceList = combineReducers( {
	canResize: simpleState( 'setSourceListCanResize', 'canResize', false ),
	isResizing: simpleState( 'setSourceListResizing', 'isResizing', false ),
	width: simpleState( 'setSourceListWidth', 'width', 330 )
} );
```